### PR TITLE
Capabilities in the dispatcher algorithm

### PR DESCRIPTION
**Pull request by jmguzik**
prowgen is coming soon as separate PR

### CODE REVIEW

File: pkg/dispatcher/config.go [hunk 1]
```diff
@@ -120,6 +154,38 @@ func (config *Config) DetermineClusterForJob(jobBase prowconfig.JobBase, path st
...
+					provider := clusterInfo.Provider
+					matchingClustersByProvider[provider] = append(matchingClustersByProvider[provider], clusterName)
+				}
+			}
+			if config.DetermineE2EByJob {
+				if cloud := config.DetermineCloudMapping(jobBase); cloud != "" {
+					if clusters, ok := matchingClustersByProvider[cloud]; ok {
+						if len(clusters) > 0 {
+							// as in other places in this file, use this method to have basic deterministic distribution
+							return api.Cluster(clusters[len(filepath.Base(path))%len(clusters)]), false, nil
```
Comment (smg247) [inline pkg/dispatcher/config.go:175]: Can we extract the "basic deterministic distribution" logic to a function?

**Comment by danilo-gemoli** [comment]
mostly ok to me

File: pkg/dispatcher/config.go (hunk 1 above)
Comment (jmguzik) [reply pkg/dispatcher/config.go:175]: We could but it would be very strange one liner as every time basically we are diving against different table len

**Comment by jmguzik** [comment]
e2e are failing on infra failure on b10
/override ci/prow/e2e

**Pull request merged**